app/controllers/api/v2/registration_controller.rb

```diff
@@ -0,0 +1,58 @@
+module Api
+  module V2
+    class RegistrationController < V2::BaseController
+      include Api::Version2
+      include Foreman::Controller::Registration
+      include Foreman::Controller::Parameters::Host
+
+      rescue_from ActiveRecord::RecordNotFound do |error|
+        logger.info "#{error.message} (#{error.class})"
+        not_found(error.message)
+      end
+
+      api :GET, '/register', N_('Render Global registration template')
+      param :organization_id, :number, desc: N_("ID of the Organization to register the host in.")
+      param :location_id, :number, desc: N_("ID of the Location to register the host in.")
+      param :hostgroup_id, :number, desc: N_("ID of the Host group to register the host in.")
+      param :operatingsystem_id, :number, desc: N_("ID of the Operating System to register the host in.")
+      def global
+        find_global_registration
+
+        unless @provisioning_template
+          not_found _('Global Registration Template with name %s defined via default_global_registration_item Setting not found, please configure the existing template name first') % Setting[:default_global_registration_item]
```

Comment:
As the endpoint renders a script, does it make sense to render error messages as a script as well?

```bash
#!/bin/sh
echo 'My error message here'
exit 1
```